platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/fingerprint/ConfigurationCacheFingerprintWriter.kt

```diff
@@ -871,11 +872,18 @@ class ConfigurationCacheFingerprintWriter(
     private
     class ProjectScopedSink(
         host: Host,
-        private val project: Path,
+        project: ProjectIdentityPath,
         private val writer: ScopedFingerprintWriter<ProjectSpecificFingerprint>
     ) : Sink(host) {
+        private
+        val projectIdentityPath = project.identityPath
+
+        init {
+            writer.write(ProjectSpecificFingerprint.ProjectIdentity(project.identityPath, project.buildPath, project.projectPath))
```

Comment:
❓ Why not carry the `ProjectIdentityPath` in `ProjectIdentity`?

Reply:
> ❓ Why not carry the `ProjectIdentityPath` in `ProjectIdentity`?

No specific reasons, except to avoid churn when a new `ProjectIdentity` type from Justin lands.